src/omnisharp/server.ts

```diff
@@ -253,63 +268,72 @@ export abstract class OmnisharpServer {
 
 	// --- start, stop, and connect
 
-	public start(solutionPath: string): Promise<void> {
-		if (!this._start) {
-			this._start = this._doStart(solutionPath);
+	private _start(launchTarget: LaunchTarget): Promise<void> {
+		const options = this._readOptions();
+		
+		let flavor: omnisharp.Flavor;
+		if (options.path !== undefined && options.usesMono === true) {
```

Comment:
That's not the intention here. Essentially, there's already an option to specify a path to a specific OmniSharp -- `csharp.omnisharp`. I've added a `csharp.omnisharpUsesMono` option that controls whether the OmniSharp specified via `csharp.omnisharp` should be launched via Mono or not. For CoreCLR, the user just has to point `csharp.omnisharp` to the CoreCLR executable and it'll be launched.